cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/externalprocess/ExternalProcessHandler.java

```diff
@@ -525,7 +491,22 @@ public boolean isNotConnected() {
         try {
             getConnection("GET", "Content-Type", "text/plain", ExternalProcessConstants.IS_CONNECTION_READY);
             if (conn.getResponseCode() < 300) {
-                return false;
+
+                // Check if it is the correct server version
+                BufferedReader br = new BufferedReader(new InputStreamReader(conn.getInputStream()));
```

Comment:
This is a memory leak. you should initialize these readers in a try-with-resource construct to make them close afterwards.